var/spack/repos/builtin/packages/steps/package.py

```diff
@@ -12,51 +14,127 @@ class Steps(CMakePackage):
     homepage = "https://groups.oist.jp/cnu/software"
     git = "https://github.com/CNS-OIST/STEPS.git"
 
-    version("3.3.0", submodules=True)
-    version("3.2.0", submodules=True)
+    maintainers("tristan0x")
+
     version("develop", branch="master", submodules=True)
+    version("4.1.1", submodules=True)
 
-    variant("native", default=True, description="Generate non-portable arch-specific code")
+    variant(
+        "codechecks",
+        default=False,
+        description="Perform additional code checks like code formatting or static analysis",
+    )
+    variant(
+        "native",
+        default=False if sys.platform == "darwin" else True,
+        description="Generate non-portable arch-specific code",
+    )
```

Comment:
In our HPC system, we have builds of this package that need to run on multiple x86 families (knl, ...). This variant is used for that, and passed as a CMake variable to the project. I don't know how it could be specified in a prettier way.

Reply:
What does the variant do, effectively? Does it just add `-march=native -mtune=native` or something more?

Reply:
:point_up: That's the only open question on my side. Other than that this PR LGTM.

Reply:
Yes, it is doing nothing more than that.

Reply:
Ok, then it should be already taken care of by our compiler wrapper. Depending on the `target` attribute, we inject the appropriate flags - so I think this variant can be removed. For a quick example of what we inject, you can try running:
```console
$ spack -d install zlib
```
At the end you should find two files similar to:
```
-rw-rw-r--  1 culpo culpo  11K apr 28 14:43 spack-cc-zlib-4kpu5pp.out.log
-rw-rw-r--  1 culpo culpo 5,2K apr 28 14:43 spack-cc-zlib-4kpu5pp.in.log
```
in your current working directory. The `*out.log` contains all the compilation flags expanded, and will look like:
```console
[cc] /usr/bin/gcc -march=icelake-client -mtune=icelake-client -c ztest21237.c
[cc] /usr/bin/gcc -march=icelake-client -mtune=icelake-client -c -fPIC -O2 ztest21237.c
...
```

Reply:
Thank you very much for your time and explanations. Indeed this `native` variant is not needed. I removed it in 7cd33e2718
Cheers